docs/user/openstack/vgpu.md

```diff
@@ -0,0 +1,456 @@
+# Installing a cluster on OpenStack with vGPU support
+
+If the underlying OpenStack deployment have proper GPU hardware installed and
+configured there is a way for passing down vGPU to the pods by using
+gpu-operator.
+
+
+## Pre-requisites
+
+The following steps are required to be checked before starting the deployment of
+OpenShift.
+
+- Appropriate hardware is installed (like [Nvidia Tesla
+  V100](https://www.nvidia.com/en-gb/data-center/tesla-v100)) on
+  the OpenStack compute node
+- NVIDIA host drivers installed and removed nouveau driver
+- Compute service installed on it and properly configured
+
+
+## Driver installation
+
+All of the examples assumes RHEL8 is used.
+
+Given, there is NVIDIA vGPU capable card installed on the machine which intended
+to have compute role, which may be confirmed by using a command which should
+display similar output:
+
+```console
+$ lspci -nn | grep -i nvidia
+3b:00.0 3D controller [0302]: NVIDIA Corporation GV100GL [Tesla V100 PCIe 16GB]
+[10de:1db4] (rev a1)
+```
+make sure to remove `nouveau` driver from loading. There might be necessary to
+add it to `/etc/modprobe.d/blacklist.conf` and/or changing
+grub config:
+
+```console
+$ sudo sed -i 's/console=/rd.driver.blacklist=nouveau console=/' /etc/default/grub
+$ sudo grub2-mkconfig -o /boot/grub2/grub.cfg
+```
+
+After that install host vGPU NVIDIA drivers:
+
+```console
+$ sudo rpm -iv NVIDIA-vGPU-rhel-8.4-510.73.06.x86_64.rpm
+```
+
+and reboot machine. Note, that drivers version may differ.
+
+After reboot, confirm there are correct drivers used:
+
+```console
+$ lsmod | grep nvidia
+nvidia_vgpu_vfio       57344  0
+nvidia              39055360  11
+mdev                   20480  2 vfio_mdev,nvidia_vgpu_vfio
+vfio                   36864  3 vfio_mdev,nvidia_vgpu_vfio,vfio_iommu_type1
+drm                   569344  4 drm_kms_helper,nvidia,mgag200
+```
+
+You can also use `nvidia-smi` tool for displaying device state.
+
+
+## OpenStack compute node
+
+There should be mediated devices populated by the driver (bus address may vary):
+
+```console
+$ ls /sys/class/mdev_bus/0000\:3b\:00.0/mdev_supported_types/
+nvidia-105  nvidia-106  nvidia-107  nvidia-108  nvidia-109  nvidia-110
+nvidia-111  nvidia-112  nvidia-113  nvidia-114  nvidia-115  nvidia-163
+nvidia-217  nvidia-247  nvidia-299  nvidia-300  nvidia-301
+```
+
+Depending of the type of workload and purchased license edition, appropriate
+types needs to be configured in `nova.conf` for compute node, i.e.:
+
+```ini
+...
+[devices]
+enabled_vgpu_types: nvidia-105
+
+...
+```
+
+After compute service restart, placement-api should report additional resources -
+command `openstack resource provider list` and `openstack resource provider inventory list <id of the main provider>`
+should display VGPU resource class available. For more information
+[navigate to OpenStack Nova docs](https://docs.openstack.org/nova/latest/admin/virtual-gpu.html).
```

Comment:
I wonder if we should point to a certain release instead of "latest".

Reply:
Right, let's make it aligned with the release of OSP we are using with OCP.